feature_layers/create-mobile-geodatabase/src/main/java/com/esri/samples/create_mobile_geodatabase/CreateMobileGeodatabaseController.java

```diff
@@ -0,0 +1,336 @@
+
+    // get the path for the geodatabase file
+    geodatabasePath = Paths.get(System.getProperty("user.dir") + "/LocationHistory.geodatabase");
+    try {
+      // delete geodatabase from previous run
+      Files.deleteIfExists(geodatabasePath);
+    } catch (IOException ioException) {
+      ioException.printStackTrace();
+    }
+
+    // create geodatabase from the specified geodatabase file path
+    var geodatabaseFuture = Geodatabase.createAsync(geodatabasePath.toString());
+    geodatabaseFuture.addDoneListener(() -> {
+      try {
+        // get the instance of the mobile geodatabase
+        geodatabase = geodatabaseFuture.get();
+
+        // create a table description to store features as map points and set non-required properties to false
+        var tableDescription = new TableDescription("LocationHistory", SpatialReferences.getWgs84(),
+          GeometryType.POINT);
+
+        // set up the fields for the table. FieldType.OID is the primary key of the SQLite table
+        var fieldDescriptionOID = new FieldDescription("oid", Field.Type.OID);
+        var fieldDescriptionText = new FieldDescription("collection_timestamp", Field.Type.TEXT);
+        tableDescription.getFieldDescriptions().addAll(List.of(fieldDescriptionOID, fieldDescriptionText));
+
+        // add a new table to the geodatabase feature table by creating one from the table description
+        var geodatabaseFeatureTableFuture = geodatabase.createTableAsync(tableDescription);
+
+        // set up the map view to display the feature layer using the loaded tableFuture geodatabase feature table
+        geodatabaseFeatureTableFuture.addDoneListener(() -> {
+          try {
+            // get the result of the loaded "LocationHistory" table
+            geodatabaseFeatureTable = geodatabaseFeatureTableFuture.get();
+            // create a feature layer for the map using the GeodatabaseFeatureTable
+            var featureLayer = new FeatureLayer(geodatabaseFeatureTable);
+            map.getOperationalLayers().add(featureLayer);
+            createGeodatabaseButton.setDisable(true);
+            closeGeodatabaseButton.setDisable(false);
+            viewTableButton.setDisable(false);
+            mapView.setDisable(false);
+            label.setText("Click map to add features");
+
+          } catch (Exception exception) {
+            new Alert(Alert.AlertType.ERROR, "Failed to get feature table result").show();
+            exception.printStackTrace();
+          }
+        });
+
+      } catch (Exception ex) {
+        new Alert(Alert.AlertType.ERROR, "Failed to get geodatabase result").show();
+        ex.printStackTrace();          }
+    });
+  }
+
+  /**
+   * Closes the geodatabase, displays its directory, and updates UI components.
+   */
+  @FXML
+  private void handleCloseGeodatabase() {
+
+    // close geodatabase, table, and clear input list
+    geodatabase.close();
+    inputs.clear();
+    tableStage.close();
```

Comment:
Found a nice little bug - if you don't view the table before closing the Geodatabase, it is not created and so a NullPointerException occurs here. So you need to add a null check:
```
    if (tableStage != null) {
      tableStage.close();
    }
```
Remember users don't necessarily click things in the right order or do what you want them to! Always worth considering when testing :)